### PR TITLE
Fix for breaking color scale in matrix legend

### PR DESCRIPTION
**Pull request by creilly8**
## Description
Fixes broken CI tests. The remaining broken test is related to a scatter commit from this morning. 

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/dom/svg.legend.js [hunk 1]
```diff
@@ -194,11 +194,20 @@ export default function svgLegend(opts) {
 		if (d.domain) {
 			colorGradientId = `sjpp-linear-gradient-${getId()}`
 			const colors = d.scale ? d.domain.map(c => d.scale(c)) : d.colors || ['white', 'grey']
+			const getData = () => {
+				if (d.domain.length > 2) {
+					if (d.domain[0] > d.minLabel) d.domain[0] = d.minLabel
```
Comment (xzhou82) [inline client/dom/svg.legend.js:199]: may avoid modifying `d.domain`?

Comment (creilly8) [reply client/dom/svg.legend.js:199]: If I avoid d.domain all together, the colors array is often longer than the data array. In [this case](http://localhost:3000/?noheader=1&mass=%7B%22dslabel%22:%22TermdbTest%22,%22genome%22:%22hg38-test%22,%22nav%22:%7B%22header_mode%22:%22hidden%22%7D,%22plots%22:%5B%7B%22chartType%22:%22hierCluster%22,%22dataType%22:%22geneExpression%22,%22terms%22:%5B%7B%22gene%22:%22TP53%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22AKT1%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22KRAS%22,%22type%22:%22geneExpression%22%7D,%7B%22gene%22:%22BCR%22,%22type%22:%22geneExpression%22%7D%5D,%22termgroups%22:%5B%7B%22name%22:%22Variables%22,%22lst%22:%5B%7B%22id%22:%22diaggrp%22%7D%5D%7D%5D%7D%5D%7D), the domain is [0, 0.5, 1] and the colors are [blue, white, red]. If I use d.domain, the ticks on the color scale won't show the true min or the max, -5 and 5. If I only use the min and the max, the color and data arrays will be different lengths. 

Do you have any thoughts?

Comment (xzhou82) [reply client/dom/svg.legend.js:199]: use domain=structuredClone(d.domain) instead of directly modifying it?

**Comment by creilly8** [comment]
Closing, taking a different direction

**Pull request closed without merging**